com.unity.netcode.gameobjects/Tests/Runtime/MultiInstanceHelpers.cs

```diff
@@ -117,7 +159,8 @@ public static void Destroy()
             // Destroy the network manager instances
             foreach (var networkManager in NetworkManagerInstances)
             {
-                Object.DestroyImmediate(networkManager.gameObject);
+                // Always destroy the root parent
```

Comment:
What if we don't want to destroy the parent?  This is making parents of NM's special.

Reply:
For MultiInstance specific testing scenarios, yes that would make this uniquely handled.
I am not aware of a condition where we want to persist the parent GameObject of a nest NetworkManager GameObject, but if we keep this approach then I could add some additional functionality.  If we go the path of just throwing an exception then the majority of the unit/integration test specific code would be removed and this would no longer be here.